src/lib/db-listeners/db-problem-documents-listener.ts

```diff
@@ -15,11 +19,20 @@ export class DBProblemDocumentsListener {
 
     // teacher user - load documents for all users
     if (user.isTeacher) {
-      const offeringUsersRef = this.db.firebase.ref(this.db.firebase.getOfferingUsersPath(user));
-      offeringUsersRef.on("value", snapshot => {
-        this.handleLoadOfferingUsersDocuments(snapshot);
+      return new Promise<void>((resolve, reject) => {
+        const offeringUsersRef = this.offeringUsersRef = this.db.firebase.ref(
+          this.db.firebase.getOfferingUsersPath(user));
+        // use once() so we are ensured that documents are set before we resolve
+        offeringUsersRef.once("value", (snapshot) => {
+          this.handleLoadOfferingUsersProblemDocuments(snapshot);
+          // We have to listen to both events because of a race condition of the documents
+          // not being set when the child is added
+          offeringUsersRef.on("child_added", this.handleLoadOfferingUserAddedOrChanged);
+          offeringUsersRef.on("child_changed", this.handleLoadOfferingUserAddedOrChanged);
```

Comment:
If I understand your concern correctly, there is another pattern for this that avoids the race condition without listening to `child_changed`: https://stackoverflow.com/a/27995609.

Reply:
The issue I was seeing is when the `child_added` events first the `documents` key isn't set yet for the user so we skip monitoring the document.  I didn't look but I assumed we were adding the document after creating the user in the code.  I had to add `child_changed` to catch the `documents` key being set.